src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php

```diff
@@ -85,6 +89,41 @@ private function getCategoriesInformation(Product $product): ProductCategoriesIn
         return new ProductCategoriesInformation($categoryIds, $defaultCategoryId);
     }
 
+    /**
+     * @param Product $product
+     *
+     * @return ProductPricesInformation
+     */
+    private function getPricesInformation(Product $product): ProductPricesInformation
+    {
+        $priceNumberFields = [
+            'price',
+            'ecotax',
+            'wholesale_price',
+            'unit_price',
+            'unit_price_ratio',
+        ];
+
+        // To make sure all values are safely converted to Number.
+        // Because casting null to string results in empty string which isn't valid to create Number and throws error.
```

Comment:
So this is to ensure that, when data is fetched from database, it is valid when put in Numbers ?

So we'll need this logic in multiple places